### ISSUE TITLE
Date::getInputFormat should support localization

### ISSUE THREAD

**Issue by Toflar** [contao/core-bundle#872]
I guess this should do it:

```diff
diff --git a/src/Resources/contao/library/Contao/Date.php b/src/Resources/contao/library/Contao/Date.php
index 4efca5e..5b2578f 100644
--- a/src/Resources/contao/library/Contao/Date.php
+++ b/src/Resources/contao/library/Contao/Date.php
@@ -297,6 +297,17 @@ class Date
 			's' => 'ss',
 		);
 
+		if (isset($GLOBALS['TL_LANG']['INPUT_FORMAT_MAPPER']))
+		{
+			foreach (array_keys($arrCharacterMapper) as $from)
+			{
+				if (isset($GLOBALS['TL_LANG']['INPUT_FORMAT_MAPPER'][$from]))
+				{
+					$arrCharacterMapper[$from] = $GLOBALS['TL_LANG']['INPUT_FORMAT_MAPPER'][$from];
+				}
+			}
+		}
+
 		$arrInputFormat = array();
 		$arrCharacters = str_split($strFormat);
 

```

**Comment by leofeyer** [comment]
I assume that I also have to create `$GLOBALS['TL_LANG']['INPUT_FORMAT_MAPPER']`? Or is everyone supposed to do so by themselves?

**Comment by Toflar** [comment]
Well, the current - hardcoded - way is the English format I suppose. You can also put the whole `$arrCharacterMapper` array to the English language files, yes.

**Comment by leofeyer** [comment]
Implemented in 907f42f9b739ff5cf4cc085c041d683825ac1d36.
